src/Refinery/KindlyTo/Transformation/TupleTransformation.php

```diff
@@ -0,0 +1,87 @@
+<?php
+declare(strict_types=1);
+/* Copyright (c) 2020 Luka K. A. Stocker, Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Refinery\KindlyTo\Transformation;
+
+use ILIAS\Refinery\DeriveApplyToFromTransform;
+use ILIAS\Refinery\Transformation;
+use ILIAS\Refinery\ConstraintViolationException;
+
+class TupleTransformation implements Transformation {
+    use DeriveApplyToFromTransform;
+
+    private $transformations;
+
+    /**
+     * @param Transformation[] $transformations;
+     */
+    public function __construct($transformations) {
+        foreach ($transformations as $transformation) {
+            if (!$transformation instanceof Transformation) {
+                $transformationClassName = Transformation::class;
+
+                throw new ConstraintViolationException(
+                    sprintf('The array must contain only "%s" instances', $transformationClassName),
+                    'not_a_transformation',
+                    $transformationClassName
+                );
+            }
+        }
+        $this->transformations = $transformations;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function transform($from) {
+        $this->ValueLength($from);
```

Comment:
We should rearrange these lines end call `ValueLength` after we ensured `$from` is a non empty array.